dockstore-webservice/src/main/java/io/dockstore/webservice/resources/SourceControlResourceInterface.java

```diff
@@ -51,16 +51,34 @@
     default Token refreshBitbucketToken(Token token, HttpClient client, TokenDAO tokenDAO, String bitbucketClientID,
         String bitbucketClientSecret) {
 
-        String url = BITBUCKET_URL + "site/oauth2/access_token";
+        String refreshUrl = BITBUCKET_URL + "site/oauth2/access_token";
+        String payload = "grant_type=refresh_token&refresh_token=" + token.getRefreshToken();
+        return refreshToken(refreshUrl, token, client, tokenDAO, bitbucketClientID, bitbucketClientSecret, payload);
+    }
+
+    /**
+     * Refreshes user's token.
+     *
+     * @param refreshUrl e.g. https://sandbox.zenodo.org/oauth/token
+     * @param token
+     * @param client
+     * @param tokenDAO
+     * @param clientID
+     * @param clientSecret
+     * @param payload e.g. "grant_type=refresh_token&refresh_token=" + token.getRefreshToken()
+     * @return the updated token
+     */
+    default Token refreshToken(String refreshUrl, Token token, HttpClient client, TokenDAO tokenDAO, String clientID,
```

Comment:
I don't know that you can make this generic. I don't think there's a guarantee that the response body for different providers will be the same, e.g., maybe somebody will call the field `access_token` and somebody else might call it `accessToken`. But maybe I'm wrong and this part of the standard.

Did you test BitBucket and verify that it still works?

Reply:
It seems to be part of the standard: https://tools.ietf.org/html/rfc6749#section-5.1

Reply:
will test

Reply:
Getting Bitbucket refresh token is working